src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -85,14 +90,18 @@ public WebSocketsTransport(HttpConnectionOptions httpConnectionOptions, ILoggerF
                     {
                         _webSocket.Options.UseDefaultCredentials = httpConnectionOptions.UseDefaultCredentials.Value;
                     }
-
-                    httpConnectionOptions.WebSocketConfiguration?.Invoke(_webSocket.Options);
                 }
 
+                httpConnectionOptions.WebSocketConfiguration?.Invoke(_webSocket.Options);
+            }
 
+            if (!isBrowser)
+            {
                 // Set this header so the server auth middleware will set an Unauthorized instead of Redirect status code
                 // See: https://github.com/aspnet/Security/blob/ff9f145a8e89c9756ea12ff10c6d47f2f7eb345f/src/Microsoft.AspNetCore.Authentication.Cookies/Events/CookieAuthenticationEvents.cs#L42
+#pragma warning disable CA1416 // Analyzer bug
```

Comment:
Is there an issue related to the analyzer bug that we can link to?

Reply:
> 5.0.100-rc.2.20472.12

I am presuming this is the SDK version? Can you try the latest RC2 NuGet package: https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5&package=Microsoft.CodeAnalysis.NetAnalyzers&protocolType=NuGet&version=5.0.0-rc2.20471.13&view=overview?

Can you also check if this repros if you replace the boolean check above with explicit `OperatingSystem.IsBrowser()` check? If that removes the false positive, then it is likely that something in flow analysis is causing the state stored in the bool to get lost before this if block. I can try to repro this later today.

Reply:
Surprsingly seems it is worked above on row 45 and 64, somehow working on off?

Reply:
![image](https://user-images.githubusercontent.com/174281/93949578-d6d9b380-fcf5-11ea-823d-9858bb59848e.png)

No good. As @buyaa-n pointed out, the same check works in the ifs above, just not after the if block on line 52